src/coreclr/pal/src/include/pal/malloc.hpp

```diff
@@ -26,40 +26,23 @@ Module Name:
 #include <stdlib.h>
 #include <new>
 
-extern "C"
-{
-    void *
-    __cdecl
-    PAL_realloc(
-        void* pvMemblock,
-        size_t szSize
-        );
-
-    void *
-    __cdecl
-    PAL_malloc(
-        size_t szSize
-        );
-
-    void
-    __cdecl
-    PAL_free(
-        void *pvMem
-        );
-}
-
 namespace CorUnix{
-
-    void *
-    InternalRealloc(
-        void *pvMemblock,
-        size_t szSize
-        );
-
-    void *
+    inline void *
     InternalMalloc(
         size_t szSize
-        );
+        )
+    {
+        void *pvMem;
+
+        if (szSize == 0)
```

Comment:
I wonder if we could fix the places where we call InternalMalloc with 0 size instead and get rid of it.

Reply:
I can look through the PAL and do that. I was trying to limit the PAL-internal changes to keep this a little smaller, but I'll take a look.